static/panes/tool.js

```diff
@@ -170,13 +192,21 @@ Tool.prototype.initArgs = function (state) {
         }
     }
 
-    if (this.stdinField) {
-        this.stdinField
+    this.monacoStdin = state.monacoStdin || false;
+    this.monacoEditorOpen = state.monacoEditorOpen || false;
+    this.monacoEditorHasBeenAutoOpened = state.monacoEditorHasBeenAutoOpened || false;
```

Comment:
I couldn't quite follow what this variable is tracking - what is "auto opened" in this context? When does the editor auto-open? (I'm sure I'm just being slow, I haven't had coffee yet! )

Reply:
No worries. If you specify a stdinHint in the config file; when you open a tool it will automatically open the stdin box.  (This was a previous patch of mine.

With the new editor, when you specify a stdinHint, we will auto-open the new editor pane.  But if you open the stdin pane; then close it; we don't want to re-open it again if you refresh the page, so we need to track if it's been auto-opened before and if so, not auto-open it again.

Reply:
Perfect! thanks